c/driver/sqlite/sqlite.c

```diff
@@ -1025,6 +1027,46 @@ AdbcStatusCode SqliteStatementInitIngest(struct SqliteStatement* stmt,
       goto cleanup;
     }
 
+    int status =
+        ArrowSchemaViewInit(&view, stmt->binder.schema.children[i], &arrow_error);
+    if (status != 0) {
+      SetError(error, "Failed to parse schema for column %d: %s (%d): %s", i,
+               strerror(status), status, arrow_error.message);
+      code = ADBC_STATUS_INTERNAL;
+      goto cleanup;
+    }
```

Comment:
for what it's worth, you could just create a view of the top-level schema and just use its children in the loop, though I suppose this gives a slightly nicer error message if one of the fields is invalid

Reply:
I tried moving `ArrowSchemaViewInit(&view, stmt->binder.schema.children[i], &arrow_error); ` out of the loop and rewriting it to `ArrowSchemaViewInit(&view, stmt->binder.schema.dictionary, &arrow_error); `but got a bunch of tests failing:
```
[  FAILED  ] 26 tests, listed below:
[  FAILED  ] SqliteConnectionTest.MetadataGetTableSchema
[  FAILED  ] SqliteConnectionTest.MetadataGetObjectsTables
[  FAILED  ] SqliteConnectionTest.MetadataGetObjectsTablesTypes
[  FAILED  ] SqliteConnectionTest.MetadataGetObjectsColumns
[  FAILED  ] SqliteStatementTest.SqlIngestInt8
[  FAILED  ] SqliteStatementTest.SqlIngestInt16
[  FAILED  ] SqliteStatementTest.SqlIngestInt32
[  FAILED  ] SqliteStatementTest.SqlIngestInt64
[  FAILED  ] SqliteStatementTest.SqlIngestUInt8
[  FAILED  ] SqliteStatementTest.SqlIngestUInt16
[  FAILED  ] SqliteStatementTest.SqlIngestUInt32
[  FAILED  ] SqliteStatementTest.SqlIngestUInt64
[  FAILED  ] SqliteStatementTest.SqlIngestFloat32
[  FAILED  ] SqliteStatementTest.SqlIngestFloat64
[  FAILED  ] SqliteStatementTest.SqlIngestString
[  FAILED  ] SqliteStatementTest.SqlIngestDate32
[  FAILED  ] SqliteStatementTest.SqlIngestTimestamp
[  FAILED  ] SqliteStatementTest.SqlIngestTimestampTz
[  FAILED  ] SqliteStatementTest.SqlIngestTableEscaping
[  FAILED  ] SqliteStatementTest.SqlIngestAppend
[  FAILED  ] SqliteStatementTest.SqlIngestErrors
[  FAILED  ] SqliteStatementTest.SqlIngestMultipleConnections
[  FAILED  ] SqliteStatementTest.SqlIngestSample
[  FAILED  ] SqliteStatementTest.SqlPrepareUpdate
[  FAILED  ] SqliteStatementTest.SqlPrepareUpdateStream
[  FAILED  ] SqliteStatementTest.Transactions
```

Example:
```
/Users/ywc/VoDa/arrow-adbc/c/validation/adbc_validation.cc:91: Failure
Value of: DoIngestSampleTable(connection, "bulk_ingest", error)
Expected: is 0 (OK)
  Actual: '\t' (9) (of type unsigned char), 9 (INTERNAL)
/Users/ywc/VoDa/arrow-adbc/c/validation/adbc_validation.cc:2172: Failure
Expected: IngestSampleTable(&connection, &error) doesn't generate new fatal failures in the current thread.
  Actual: it does.
[  FAILED  ] SqliteStatementTest.Transactions (0 ms)
```

Reply:
Why are you initializing it against the dictionary...? You would just initialize it against `stmt->binder.schema`.

Reply:
Ah, ignore this - SchemaView isn't nested, only ArrayView is, so you can't move it out of here.